orchestrator/graphql/resolvers/product.py

```diff
@@ -0,0 +1,65 @@
+from typing import Union
+
+import structlog
+from graphql import GraphQLError
+
+from orchestrator.db.filters import CallableErrorHander, Filter
+from orchestrator.db.filters.product import filter_products
+from orchestrator.db.models import ProductTable
+from orchestrator.db.range.range import apply_range_to_query
+from orchestrator.db.sorting.product import sort_products
+from orchestrator.db.sorting.sorting import Sort
+from orchestrator.graphql.pagination import Connection, PageInfo
+from orchestrator.graphql.schemas.product import ProductType
+from orchestrator.graphql.types import CustomInfo, GraphqlFilter, GraphqlSort
+
+logger = structlog.get_logger(__name__)
+
+
+def handle_product_error(info: CustomInfo) -> CallableErrorHander:
+    def _handle_product_error(message: str, **kwargs) -> None:  # type: ignore
+        logger.debug(message, **kwargs)
+        extra_values = dict(kwargs.items()) if kwargs else {}
+        info.context.errors.append(GraphQLError(message=message, path=info.path, extensions=extra_values))
+
+    return _handle_product_error
+
+
+async def resolve_products(
+    info: CustomInfo,
+    filter_by: Union[list[GraphqlFilter], None] = None,
+    sort_by: Union[list[GraphqlSort], None] = None,
+    first: int = 10,
+    after: int = 0,
+) -> Connection[ProductType]:
+    _error_handler = handle_product_error(info)
+
+    _range: Union[list[int], None] = [after, after + first] if after is not None and first else None
+    pydantic_filter_by: list[Filter] = [item.to_pydantic() for item in filter_by] if filter_by else []
```

Comment:
Maybe just 'filter_by' and 'sort_by' as names. Not sure the 'pydantic_' is relevant in the naming.

Reply:
done the same way as in processes, I didn't want to overwrite the existing variable.